### ISSUE TITLE
Details about MC dropout

### ISSUE THREAD

**Issue by haruishi43** [Jianf-Wang/NP-SemiSeg#3]
Thank you for your amazing work in the probabilistic approach for SSL.
I would like to know more about the MC dropout baseline in your paper:

1. Are the uncertainty estimations obtained from MC dropout used to supervise the model during training? If so, how are they being used? It seems like the uncertainty estimations for MC dropout are only used to evaluate PAvPU.
2. Do you open source the relevant code for MC dropout? If the MC dropout is only used to evaluate uncertainty estimation and not to supervised the model, I understand you omitting the codebase.

**Comment by Jianf-Wang** [comment]
Hi, thanks for your interest in this work.

1. No, they are only used for evaluating PAvPU, in order to compare the performance of MC dropout with that of NPs in terms of uncertainty quantification.
2. I did not open source the MC dropout part since MC dropout is inserted in convolution layers in the backbone, and I implemented this with additional model files. To make the code clearer, I omitted those files. But I believe it is easy to implement them by yourself, just by adding some dropout layers.

**Comment by Jianf-Wang** [comment]
@haruishi43, Hi, thanks for your interest in this work.

1. No, they are only used for evaluating PAvPU, in order to compare the performance of MC dropout with that of NPs in terms of uncertainty quantification.

2. I did not open source the MC dropout part since MC dropout is inserted in convolution layers in the backbone, and I implemented this with additional model files. To make the code clearer, I omitted those files. But I believe it is easy to implement them by yourself, just by adding some dropout layers.

**Comment by haruishi43** [comment]
@Jianf-Wang thanks for your response. 
I've just recently began studying stochastic approach for uncertainty estimation, so I was just curious how you implemented it. If I'm not mistaken, you would set the model to `.train()` mode so that dropout is enabled for inference, but that would result in batchnorm statistics being updated during eval, so did you make sure that the statistics are frozen? Also, did you use `T=5` for MC dropout for your evaluation for tables 1~4?

**Comment by Jianf-Wang** [comment]
You may consider using torch.nn.functional.Dropout, and setting a parameter for your forward function, controlling the dropout function. Yes, T=5.
